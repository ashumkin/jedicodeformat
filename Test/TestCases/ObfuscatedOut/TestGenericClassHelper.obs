program testgenericclasshelper;{$APPTYPE CONSOLE}uses sysutils;type tgenerique<t
> =class champ:t end;tgeneriqueint=tgenerique<integer>;thelpergeneriqueint=class 
helper for tgeneriqueint procedure test;end;tgeneriquestring=tgenerique<string>;procedure
thelpergeneriqueint.test;begin writeln('Procédure fournie par un assistant de classe.'
);end;var assistant:tgeneriqueint;assistant2:tgeneriquestring;begin try assistant
:=tgeneriqueint.create;assistant.test;except on e:exception do writeln(e.classname
,': ',e.message);end;readln;end. 