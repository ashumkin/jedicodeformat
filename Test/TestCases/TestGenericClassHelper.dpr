program TestGenericClassHelper;

{$APPTYPE CONSOLE}

{ AFS December 2007 

  This code compiles, but is not semantically meaningfull.
  It is test cases for the code-formating utility

  Test new generics syntax - code from TridenT
}

uses
  SysUtils;

type
  TGenerique<T>  = class
   Champ: T
  end;

  {
   E2003 Identificateur non déclaré : 'T'
  THelperGenerique=Class Helper for TGenerique<T>
   Procedure Test;
  End;

   E2508 Les paramètres de type ne sont pas autorisés sur ce type
  THelperGenerique<T>=Class Helper for TGenerique<T>
   Procedure Test;
  End;
 }
  TGeneriqueInt=TGenerique<Integer>;
  THelperGeneriqueInt=Class Helper for TGeneriqueInt
   Procedure Test;
  End;

  TGeneriqueString=TGenerique<String>;

Procedure THelperGeneriqueInt.Test;
begin
  Writeln('Procédure fournie par un assistant de classe.');
end;


var Assistant :TGeneriqueInt;
    Assistant2 :TGeneriqueString;
begin
  try
   Assistant:=TGeneriqueInt.Create;
   Assistant.Test;
   //Assistant2:=TGeneriqueString.Create;
   //Assistant2.Test;   //E2003 Identificateur non déclaré : 'Test'
  except
    on E:Exception do
      Writeln(E.Classname, ': ', E.Message);
  end;
  Readln;
end.
