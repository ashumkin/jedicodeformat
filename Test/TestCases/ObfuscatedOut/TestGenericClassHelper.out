program testgenericclasshelper;{$APPTYPE CONSOLE}

uses
  SysUtils;

type
  tgenerique<t> = class
    champ: t
  end;
  tgeneriqueint = tgenerique<integer>;

  thelpergeneriqueint = class helper for tgeneriqueint
    procedure test;
  end;
  tgeneriquestring = tgenerique<string>;

  procedure thelpergeneriqueint.test;
  begin
    writeln('Procédure fournie par un assistant de classe.');
  end;

var
  assistant:  tgeneriqueint;
  assistant2: tgeneriquestring;
begin
  try
    assistant := tgeneriqueint.Create;
    assistant.test;
  except
    on e: Exception do
      writeln(e.ClassName, ': ', e.message);
  end;
  readln;
end.
 